Aptoide/aptoide-client-v8
# [ASV-594] Logout delete account users

**What does this PR do?**

When a user deletes his account, his tokens are invalidated. Then when he makes any request, it will return an error of an invalid access token. Since the refresh token is also invalid, he is not able to get a new access token, so the requests will fail and the user will be stuck. In this PR, we retry to refresh the access token 3 times  and if it is still returning an error, then we logout the user. When we logout the user we must make sure that not only the account was removed, but also that the avatar on the toolbar changes to the default one.

**Database changed?**

   No

**Where should the reviewer start?**

- [ ] RefreshTokenInvalidator.java

**How should this be manually tested?**

Create an account, delete that account. Close the app or go another view where there is a request needed to be done. Wait until you receive the confirmation email for the account deletion and move in the app to somewhere where it is needed to do some requests. Those requests will fail because they will be done with an invalid access token. We will try to refresh the token 3 times ( by calling the oAuth2Authentication ) and if after the 3rd attempt the token is still invalid, we will logout the user.
During the development, after deleting the account i would close aptoide and then when i got the email i would re open it, just because on the home you can see everything (there are multiple requests and there is the user avatar on the toolbar.

**What are the relevant tickets?**

  Tickets related to this pull-request: [ASV-594](https://aptoide.atlassian.net/browse/ASV-594)

**Questions:**

   Does this add new dependencies which need to be added? (Eg. new libs, new keys, etc.) 




**Code Review Checklist**
